embassy-stm32/src/wdg/mod.rs

```diff
@@ -0,0 +1,52 @@
+use core::marker::PhantomData;
+
+use embassy::util::Unborrow;
+use embassy_hal_common::unborrow;
+
+use stm32_metapac::iwdg::vals::Key;
+
+pub use stm32_metapac::iwdg::vals::Pr as Prescaler;
+
+pub struct IndependentWatchdog<'d, T: Instance> {
+    wdg: PhantomData<&'d mut T>,
+}
+
+impl<'d, T: Instance> IndependentWatchdog<'d, T> {
+    pub fn new(_instance: impl Unborrow<Target = T>, presc: Prescaler) -> Self {
+        unborrow!(_instance);
+
+        let wdg = T::regs();
+        unsafe {
+            wdg.kr().write(|w| w.set_key(Key::ENABLE));
+            wdg.pr().write(|w| w.set_pr(presc));
+        }
+
+        IndependentWatchdog {
+            wdg: PhantomData::default(),
+        }
+    }
+
+    pub unsafe fn unleash(&mut self) {
+        T::regs().kr().write(|w| w.set_key(Key::START));
+    }
+
+    pub unsafe fn feed(&mut self) {
```

Comment:
I've seen the term `pet` being used in nRF HAL, would be good to standardize on the same terms